From ea41179dab5df144d2ec137c94dcbe171e427728 Mon Sep 17 00:00:00 2001
From: Iain Sandoe <iain@sandoe.co.uk>
Date: Wed, 18 Sep 2024 17:46:32 +0100
Subject: [PATCH 1/3] Darwin: Allow for as versions that need '-' for std in.

Recent versions of Xcode as require a dash to read from standard
input.  We can use this on all supported OS versions so make it
unconditional.  Patch from Mark Mentovai.

gcc/ChangeLog:

	* config/darwin.h (AS_NEEDS_DASH_FOR_PIPED_INPUT): New.

Signed-off-by: Iain Sandoe <iain@sandoe.co.uk>
---
 gcc/config/darwin.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git gcc/config/darwin.h gcc/config/darwin.h
index 7a53a1688a7d..5370511bec21 100644
--- gcc/config/darwin.h
+++ gcc/config/darwin.h
@@ -651,6 +651,8 @@ extern GTY(()) int darwin_ms_struct;
 #define ASM_OPTIONS "%{v} %{w:-W} %{I*}"
 #endif
 
+#define AS_NEEDS_DASH_FOR_PIPED_INPUT
+
 /* Default Darwin ASM_SPEC, very simple. */
 #define ASM_SPEC \
 "%{static} -arch %(darwin_arch) " \
-- 
2.46.2

